### PR TITLE
[Pet Set] Fix useless pointer array copy

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

addresses #31687

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31689)

### CODE REVIEW

**Comment by m1093782566** [comment]
@bprashanth PTAL

Thanks!

File: pkg/controller/petset/pet_set.go [hunk 1]
```diff
@@ -222,12 +222,7 @@ func (psc *PetSetController) getPodsForPetSet(ps *apps.PetSet) ([]*api.Pod, erro
 	if err != nil {
 		return []*api.Pod{}, err
 	}
-	// TODO: Do we need to copy?
-	result := make([]*api.Pod, 0, len(pods))
-	for i := range pods {
-		result = append(result, &(*pods[i]))
```
Comment (bprashanth) [inline pkg/controller/petset/pet_set.go]: actually the intent is to protect the list (i.e if Pods.List() returns a saved list every time, and we just return that, doing returnedList[1]=*Pod{} would reflect as returnedList[1] in the callers), not the elements in the list (that's what the todo is for).

Comment (m1093782566) [inline pkg/controller/petset/pet_set.go]: Sorry, I am afraid I didn't catch your meaning.

* I think petset controller only use the returned `Pods.List()` for reading and won't affect the elements of original list. So I remove the following code block directly.


```
    result := make([]*api.Pod, 0, len(pods))
	for i := range pods {
		result = append(result, &(*pods[i]))
	}
```

* Do we need to copy the elements of the original list? Like,
```
    result := make([]*api.Pod, 0, len(pods))
	for i := range pods {
        p := *pods[i]
		result = append(result, &p)
	}
	return result, nil
```

* I just can't understand the real purpose of copying pointers from an array to another, to protect the original list?

Comment (bprashanth) [inline pkg/controller/petset/pet_set.go]: https://play.golang.org/p/c9-sxjd19q

Comment (m1093782566) [inline pkg/controller/petset/pet_set.go]: I think I understand your meaning, you are right :) 

Thanks!

**Pull request closed without merging**